tests/testsuite/features_namespaced.rs

```diff
@@ -1216,3 +1216,81 @@ Caused by:
         )
         .run();
 }
+
+#[cargo_test]
+fn dep_feature_when_hidden() {
+    // Checks for behavior with dep:bar and bar/feat syntax when there is no
+    // `bar` feature.
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+
+                [dependencies]
+                bar = { path = "bar", optional = true }
+
+                [features]
+                f1 = ["dep:bar"]
+                f2 = ["bar/bar_feat"]
```

Comment:
I think I would have preferred for this to be `dep:bar/bar_feat` for improved consistency but that is already out the window because `bar?/bar_feat` already works.